libs/sdk-backend-tiger/src/convertors/toBackend/afm/FilterConverter.ts

```diff
@@ -145,49 +164,138 @@ export function convertMeasureValueFilter(filter: IMeasureValueFilter): MeasureV
         const { operator, from: originalFrom, to: originalTo, treatNullValuesAs } = condition.range;
         return {
             rangeMeasureValueFilter: {
-                measure: toMeasureValueFilterMeasureQualifier(measureValueFilter.measure),
+                measure: toMeasureValueFilterMeasureQualifier(filteredMeasureId),
                 // Operator has same values, we only need type assertion
                 operator: operator as RangeMeasureValueFilterBodyOperatorEnum,
                 // make sure the boundaries are always from <= to, because tiger backend cannot handle from > to in a user friendly way
                 // this is effectively the same behavior as in bear
                 from: Math.min(originalFrom, originalTo),
                 to: Math.max(originalFrom, originalTo),
                 treatNullValuesAs,
+                ...applyOnResultProp,
             },
         };
     }
 
     return null;
 }
 
-export function convertRankingFilter(filter: IRankingFilter): RankingFilter {
+export function convertRankingFilter(
+    computeRatioMeasureNumerators: Map<Identifier, IMeasure<IMeasureDefinition>>,
+    filter: IRankingFilter,
+): RankingFilter {
     const { measure, attributes, operator, value } = filter.rankingFilter;
     const dimensionalityProp = attributes
         ? { dimensionality: attributes.map(toRankingFilterDimensionalityIdentifier) }
         : {};
+    const [filteredMeasureId, applyOnResult] = getActualFilteredMeasureInfo(
+        computeRatioMeasureNumerators,
+        measure,
+    );
+    const applyOnResultProp = applyOnResult == undefined ? {} : { applyOnResult };
+
     return {
         rankingFilter: {
-            measures: [toMeasureValueFilterMeasureQualifier(measure)],
+            measures: [toMeasureValueFilterMeasureQualifier(filteredMeasureId)],
             ...dimensionalityProp,
             operator: operator as RankingFilterBodyOperatorEnum,
             value,
+            ...applyOnResultProp,
         },
     };
 }
 
-export function convertVisualizationObjectFilter(
-    filter: IMeasureFilter | IFilter,
-): FilterDefinition | FilterDefinitionForSimpleMeasure | null {
+/**
+ * Return the actual measure which should be used for measure based filtering, together with the info how the
+ * corresponding filter should be applied (`undefined` means default application).
+ */
+function getActualFilteredMeasureInfo(
+    computeRatioMeasureNumerators: ComputeRatioMeasureNumerators,
+    measure: ObjRefInScope,
+): [ObjRefInScope, boolean?] {
+    const updatedFilteredMeasure = isLocalIdRef(measure)
+        ? computeRatioMeasureNumerators.get(measure.localIdentifier)
+        : undefined;
+    return updatedFilteredMeasure == undefined
+        ? [measure, undefined]
+        : [localIdRef(updatedFilteredMeasure.measure.localIdentifier), false];
```

Comment:
please use getter function instead of `updatedFilteredMeasure.measure.localIdentifier`: `measureLocalId(updatedFilteredMeasure)`

Reply:
fixed